code/modules/client/preferences/_preferences.dm

```diff
@@ -598,28 +594,94 @@ GLOBAL_LIST_INIT(name_adjustments, list())
 				if(job.whitelist_req && (!user.client.whitelisted()))
 					HTML += "<font color=#6183a5>[used_name]</font></td> <td> </td></tr>"
 					continue
-
-			if(get_playerquality(user.ckey) < job.min_pq)
-				HTML += "<font color=#a36c63>[used_name] (Min PQ: [job.min_pq])</font></td> <td> </td></tr>"
-				continue
-			if(length(job.allowed_ages) && !(user.client.prefs.age in job.allowed_ages))
-				HTML += "<font color=#a36c63>[used_name]</font></td> <td> </td></tr>"
-				continue
 			if((length(job.allowed_races) && !(user.client.prefs.pref_species.id in job.allowed_races)) || \
 				(length(job.blacklisted_species) && (user.client.prefs.pref_species.id in job.blacklisted_species)))
 				if(!(user.client.has_triumph_buy(TRIUMPH_BUY_RACE_ALL)))
-					HTML += "<font color=#a36c63>[used_name]</font></td> <td> </td></tr>"
+					var/races_text = jointext(job.allowed_races, ", ")
+					HTML += {"
+							[used_name]
+						</td>
+						<td>
+							<div class='tutorialhover'>
+								<font color=#a36c63>\[SPECIES LOCK\]</font>
+								<span class='tutorial'><b>Species Needed:</b><br>[races_text]</span>
+							</div>
+						</td>
+						</tr>
+					"}
 					continue
-			if(length(job.allowed_patrons) && !(user.client.prefs.selected_patron.type in job.allowed_patrons))
-				HTML += "<font color=#a36c63>[used_name]</font></td> <td> </td></tr>"
+			if(length(job.allowed_ages) && !(user.client.prefs.age in job.allowed_ages))
+				var/ages_text = jointext(job.allowed_ages, ", ")
+				HTML += {"
+						[used_name]
+					</td>
+					<td>
+						<div class='tutorialhover'>
+							<font color=#a36c63>\[AGE LOCK\]</font>
+							<span class='tutorial'><b>Ages Needed:</b><br>[ages_text]</span>
+						</div>
+					</td>
+					</tr>
+				"}
 				continue
 			if(length(job.allowed_sexes) && !(user.client.prefs.gender in job.allowed_sexes))
-				HTML += "<font color=#a36c63>[used_name]</font></td> <td> </td></tr>"
+				var/sexes_text = jointext(job.allowed_sexes, ", ")
+				HTML += {"
+						[used_name]
+					</td>
+					<td>
+						<div class='tutorialhover'>
+							<font color=#a36c63>\[SEX LOCK\]</font>
+							<span class='tutorial'><b>Sexes Needed:</b><br>[sexes_text]</span>
+						</div>
+					</td>
+					</tr>
+				"}
+				continue
+			if(length(job.allowed_patrons) && !(user.client.prefs.selected_patron.type in job.allowed_patrons))
+				var/list/patron_list = list()
+				for(var/mult_patron in job.allowed_patrons)
+					var/datum/patron/P = new mult_patron
+					patron_list += (P.display_name ? P.display_name : P.name)
+					qdel(P)
+				var/patron_text = jointext(patron_list, ", ")
+				HTML += {"
+						[used_name]
+					</td>
+					<td>
+						<div class='tutorialhover'>
+							<font color=#a36c63>\[PATRON LOCK\]</font>
+							<span class='tutorial'><b>Patron Needed:</b><br>[patron_text]</span>
+						</div>
+					</td>
+					</tr>
+				"}
```

Comment:
This is terrible, this should be handled by a proc that can return the reason in the tutorial tooltip

Reply:
What is the point of a proc that is going to be used twice, it will need a variant since some are instanced and others are jus the raw value